src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -169,6 +169,229 @@ public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
         }
     }
 
+    public sealed class VectorOverloader : IOverloader
+    {
+        private static readonly Regex VectorNameMatch = new Regex("([1-4])([f|d|h|i])v$", RegexOptions.Compiled);
+        private static readonly Regex MatrixNameMatch = new Regex("Matrix([1-4])(?:x([1-4]))?([f|d])v$", RegexOptions.Compiled);
```

Comment:
This can be changed to `"Matrix([1-4](?:x[1-4])?)([fd])v$"` together with the comment about doing the matrix parsing differently.

Reply:
If we are not changing this regex the `[f|d]` should at least be changed to `[fd]`.